recipes/fruit/all/patches/0003-supports-cmake-multi-configuration-generator-3.7.0.patch

```diff
@@ -0,0 +1,91 @@
+diff --git a/CMakeLists.txt b/CMakeLists.txt
+index 03444f5..46d3918 100644
+--- a/CMakeLists.txt
++++ b/CMakeLists.txt
+@@ -14,15 +14,8 @@ set(CMAKE_CXX_STANDARD_REQUIRED ON)
+ # CMake on OSX likes to see this set explicitly, otherwise it outputs a warning.
+ set(CMAKE_MACOSX_RPATH 1)
+ 
+-if(NOT "${CMAKE_BUILD_TYPE}" MATCHES "^(Debug|Release|RelWithDebInfo|MinSizeRel)$")
+-  message(FATAL_ERROR "Please re-run CMake, specifying -DCMAKE_BUILD_TYPE=Debug , -DCMAKE_BUILD_TYPE=Release , -DCMAKE_BUILD_TYPE=RelWithDebInfo or -DCMAKE_BUILD_TYPE=MinSizeRel .")
+-endif()
+-
+ option(BUILD_SHARED_LIBS "Build shared library" ON)
+ 
+-# The Visual Studio CMake generators default to multiple configurations, but Fruit doesn't support multi-configuration build directories.
+-set(CMAKE_CONFIGURATION_TYPES "${CMAKE_BUILD_TYPE}")
+-
+ set(FRUIT_ADDITIONAL_CXX_FLAGS "" CACHE STRING "Additional CXX compiler flags." FORCE)
+ 
+ set(FRUIT_ADDITIONAL_COMPILE_FLAGS "${FRUIT_ADDITIONAL_CXX_FLAGS}")
+@@ -36,23 +29,18 @@ if(NOT "${CMAKE_CXX_COMPILER_ID}" MATCHES "^(GNU|Clang|AppleClang|MSVC)$")
+   #   - https://software.intel.com/en-us/forums/intel-c-compiler/topic/606049
+ endif()
+ 
+-option(FRUIT_ADD_WNO_UNKNOWN_WARNING_OPTION "Add -Wno-unknown-warning-option to the compiler options for GCC, Clang, ICC and AppleClang" ON)
++set(FRUIT_ADDITIONAL_COMPILE_FLAGS_GNU ${FRUIT_ADDITIONAL_COMPILE_FLAGS} -W -Wall -Wno-unknown-warning-option -Wno-missing-braces)
++set(FRUIT_ADDITIONAL_COMPILE_FLAGS_Clang ${FRUIT_ADDITIONAL_COMPILE_FLAGS} -W -Wall -Wno-unknown-warning-option -Wno-missing-braces)
```

Comment:
Why do we need play with the warnings? does this fail the build?

Reply:
I will answer since I was the first to make this patch.
This flag was set in the original fruit repository. I don't know why this flag is needed.
Nevertheless, removing the warnings may make it more difficult to investigate defects because the warnings that library developers expect to be displayed at build time will no longer be displayed, so I decided to keep the flags.

TIP: The original `CMakeLists.txt` did not support CMake multi-configuration-generator, so it has been modified.

Reply:
Ahh then i'd highly recommend making a PR against upstream to contribute back this improvement  :)

Adding support for multi-config generators is fine :)

Reply:
I remember not making PR because the maintainer was busy at the time and did not seem to have time to maintain it.
The maintainer's response has been quick, so they may be able to check PR now.

I am currently away from the C/C++ project and don't have much time available, so it would be great if someone could create PR.

Reply:
Open PR now: google/fruit#149